packages/async-rewriter/src/async-writer-babel.ts

```diff
@@ -1,167 +1,179 @@
 /* eslint no-console:0 */
 import * as babel from '@babel/core';
 // import printAST from 'ast-pretty-print';
+import SymbolTable from './symbol-table';
 
-export default (symbols): any => {
-  const debug = (str, type?): void => {
-    str = `  ${str}${type === undefined ? '' : ` ==> ${type}`}`;
-    // console.log(str);
-  };
+const debug = (str, type?): void => {
+  str = `  ${str}${type === undefined ? '' : ` ==> ${type}`}`;
+  // console.log(str);
+};
 
-  const inferTypeVisitor = {
-    Identifier: {
-      exit(path): void {
-        const id = path.node.name;
-        let sType = symbols.lookup(id);
-        if (typeof sType === 'string') {
-          sType = symbols.types[sType];
+export default class AsyncWriter {
+  public symbols: SymbolTable;
```

Comment:
This is public in case we need to access the symbol table from elsewhere, potentially for debugging or logging.